### PR TITLE
cleanup Head::simulate() code

### PR DESCRIPTION
**Pull request by AndrewMeadows**
* move MyAvatar-specific code in Head::simulate() to MyHead::simulate()
* move AvatarMotionState back into interface codebase

### CODE REVIEW

**Comment by AndrewMeadows** [comment]
While looking for a bug I got lost and cleaned up a bunch of code that no longer belongs in Head::simulate().  The bug was eventually found and fixed by someone else.  Although these changes don't fix any bugs I decided I would bundle them into a PR rather than throw them away.

**To test:**

(1) Login UserA.
(2) Go into mirror mode and speak into the mic to make the face blendshapes change (e.g. make the mouth move).
--> verify that the mouth still animates when the audio is loud
--> verify that the eyes blink after the audio input goes quiet

(3) Login a second UserB
(4) Stand UserA and UserB face to face so that each can see the other.
(5) Have each user speak into the mic
--> verify that the speaker's mouth moves when the audio is loud
--> verify that the speaker's eyes blink after they stop talking

(6) If you have FaceTracker/EyeTracker technology available then set that up for one of the users and verify that their face and eyes move correctly according to the tracker input.

File: libraries/avatars-renderer/src/avatars-renderer/Head.cpp [hunk 1]
```diff
@@ -63,116 +58,114 @@ void Head::simulate(float deltaTime) {
...
+    _eyePosition = 0.5f * (_leftEyePosition + _rightEyePosition);
+}
 
-    _eyePosition = calculateAverageEyePosition();
+void Head::simulate(float deltaTime) {
+    computeAudioLoudness(deltaTime);
+    computeFaceMovement(deltaTime);
+    computeEyeMovement(deltaTime);
+    computeEyePosition();
 }
```
Comment (AndrewMeadows) [inline libraries/avatars-renderer/src/avatars-renderer/Head.cpp]: By default otherAvatar's don't have Face/EyeTracker technology available so the check for that was moved out of `Head` class and into `MyHead`.  Instead we group each type of simulation operation into its own function.

If you look in `MyHead::simulate()` you'll see that we now check for the Face/EyeTracker tech and if present we gather the data, else we call the relevant base methods from the Head.

File: libraries/avatars/src/HeadData.cpp [hunk 2]
```diff
@@ -28,12 +28,6 @@ HeadData::HeadData(AvatarData* owningAvatar) :
     _basePitch(0.0f),
     _baseRoll(0.0f),
     _lookAtPosition(0.0f, 0.0f, 0.0f),
-    _isFaceTrackerConnected(false),
-    _isEyeTrackerConnected(false),
-    _leftEyeBlink(0.0f),
-    _rightEyeBlink(0.0f),
-    _averageLoudness(0.0f),
-    _browAudioLift(0.0f),
```
Comment (AndrewMeadows) [inline libraries/avatars/src/HeadData.cpp]: These now have default initial values where they are declared in the class.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 04827c3a2a39544cdddda68d0bde52d9beb2a512 available at: https://gist.github.com/3f17dcacbee8ac8bc1343b60a27b34da<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10445-04827c3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10445-04827c3.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a43447310df5174942bc864bd1ed8333568ae2a9 available at: https://gist.github.com/3f17dcacbee8ac8bc1343b60a27b34da<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10445-a434473.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10445-a434473.dmg

**Comment by OmegaHeron** [comment]
Testing now

**Comment by OmegaHeron** [comment]
Tested Windows 10 and Ubuntu 16.04
QA Pass

Test 1...2 Pass - Mouth moves, blinks on voice to !voice transit.
Test 3...5 Pass - Each avatar can see own mouth move and blink on silence in mirror mode and in respect to each other.
Test 6 - Not tested, no capability for.
Test 7 - Pass.  Avatars collide with each other and object with collision shapes in scene.

**Pull request merged**